File: TP06/bingo.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

int main() {
    int numCartelas, numerosPorCartela, numerosSorteados;
    scanf("%d %d %d", &numCartelas, &numerosPorCartela, &numerosSorteados);
    
    int **cartelas = (int**)malloc(numCartelas * sizeof(int*));
    for (int i = 0; i < numCartelas; i++) {
        cartelas[i] = (int*)malloc(numerosPorCartela * sizeof(int));
    }
    
    int *acertos = (int*)malloc(numCartelas * sizeof(int));
    for (int i = 0; i < numCartelas; i++) {
        acertos[i] = 0;
    }
    
    for (int i = 0; i < numCartelas; i++) {
        for (int j = 0; j < numerosPorCartela; j++) {
            scanf("%d", &cartelas[i][j]);
        }
    }
    
    int *sorteados = (int*)malloc(numerosSorteados * sizeof(int));
    for (int i = 0; i < numerosSorteados; i++) {
        scanf("%d", &sorteados[i]);
    }
    
    bool achouVencedor = false;
    bool primeiroVencedor = true;
    
    for (int i = 0; i < numerosSorteados && !achouVencedor; i++) {
        int numeroAtual = sorteados[i];
        
        for (int c = 0; c < numCartelas; c++) {
            for (int k = 0; k < numerosPorCartela; k++) {
                if (cartelas[c][k] == numeroAtual) {
                    acertos[c]++;
                    break;
                }
            }
        }
        
        for (int c = 0; c < numCartelas; c++) {
            if (acertos[c] == numerosPorCartela) {
                achouVencedor = true;
                
                if (primeiroVencedor) {
                    printf("%d", c + 1);
                    primeiroVencedor = false;
                } else {
                    printf(" %d", c + 1);
                }
            }
        }
    }
    
    printf("\n");
    
    for (int i = 0; i < numCartelas; i++) {
        free(cartelas[i]);
    }
    free(cartelas);
    free(acertos);
    free(sorteados);
    
    return 0;
}

File: TP06/bingo.java
import java.util.Scanner;

public class bingo {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int numCartelas = sc.nextInt();
        int numerosPorCartela = sc.nextInt();
        int numerosSorteados = sc.nextInt();

        int[][] cartelas = new int[numCartelas][numerosPorCartela];
        int[] acertos = new int[numCartelas];

        for (int i = 0; i < numCartelas; i++) {
            for (int j = 0; j < numerosPorCartela; j++) {
                cartelas[i][j] = sc.nextInt();
            }
        }

        int[] sorteados = new int[numerosSorteados];
        for (int i = 0; i < numerosSorteados; i++) {
            sorteados[i] = sc.nextInt();
        }

        String saida = "";
        boolean achouVencedor = false;

        for (int i = 0; i < numerosSorteados && !achouVencedor; i++) {
            int numeroAtual = sorteados[i];

            for (int c = 0; c < numCartelas; c++) {
                for (int k = 0; k < numerosPorCartela; k++) {
                    if (cartelas[c][k] == numeroAtual) {
                        acertos[c]++;
                        break;
                    }
                }
            }

            for (int c = 0; c < numCartelas; c++) {
                if (acertos[c] == numerosPorCartela) {
                    achouVencedor = true;

                    if (saida.isEmpty()) {
                        saida += (c + 1);
                    } else {
                        saida += " " + (c + 1);
                    }
                }
            }
        }

        System.out.println(saida);
        sc.close();
    }
}